src/node/src/indexer_impl.rs

```diff
@@ -99,77 +97,132 @@ impl IndexerImpl {
+                    Ok((data, data_type, account_id)) => match data_type {
+                        PayloadType::DatabasePayload => {
+                            match Self::parse_database_mutation(data.as_ref()) {
+                                Ok(dm) => {
+                                    let action = DatabaseAction::from_i32(dm.action);
+                                    info!(
+                                        "Add account: {}, mutation : {:?}, tx: {} into pending queue",
+                                        account_id.addr.to_hex(), action, tx_id.to_base64());
+                                    database_mutations.push((account_id.addr, dm, tx_id));
                                 }
-                                _ => {
-                                    info!("Skip other payload type");
+                                Err(e) => {
+                                    let msg = format!("{e}");
+                                    return Err(Status::internal(msg));
                                 }
-                            },
-                            Err(e) => {
-                                warn!("invalid write request: {:?}", e);
-                                return Err(Status::internal(format!(
-                                    "invalid write request: {:?}",
-                                    e
-                                )));
                             }
-                        },
-                        Err(e) => {
-                            warn!("invalid write request: {:?}", e);
-                            return Err(Status::internal(format!(
-                                "invalid write request: {:?}",
-                                e
-                            )));
                         }
+                        PayloadType::QuerySessionPayload => {
+                            debug!("[Indexer] Skip QuerySessionPayload");
+                        }
+                        PayloadType::MintCreditsPayload => {
+                            debug!("[Indexer] Skip MintCreditsPayload");
+                        }
+                        _ => {
+                            debug!("[Indexer] Skip other payload type");
+                        }
+                    },
+                    Err(e) => {
+                        warn!("[Indexer] invalid write request: {:?}", e);
+                        return Err(Status::internal(format!("invalid write request: {:?}", e)));
                     }
+                },
+                Err(e) => {
+                    warn!("[Indexer] invalid write request: {:?}", e);
+                    return Err(Status::internal(format!("invalid write request: {:?}", e)));
                 }
             }
-            _ => {}
         }
         Ok(())
     }
+    async fn apply_database_mutations(
+        auth_store: &mut AuthStorage,
+        pending_databases: &mut Vec<(AccountAddress, DatabaseMutation, TxId)>,
```

Comment:
remove the `mut`

Reply:
resolve in PR https://github.com/dbpunk-labs/db3/pull/453